src/resmom/mom_main.c

```diff
@@ -6993,31 +6997,32 @@ mom_over_limit(job *pjob)
 				svr_resc_size);
 			assert(rd != NULL);
 			preswalltime = find_resc_entry(at, rd);
-			assert(preswalltime != NULL);
-			walltime_sum = preswalltime->rs_value.at_val.at_long;
-			if (walltime_sum > average_trialperiod) {
-				rd = find_resc_def(svr_resc_def, "cput",
-					svr_resc_size);
-				assert(rd != NULL);
-				prescput = find_resc_entry(at, rd);
-				assert(prescput != NULL);
-				cput_sum = prescput->rs_value.at_val.at_long;
-				/* "value" is from ncpus */
-				if (((double)cput_sum/(double)walltime_sum) >
-					(value*average_cpufactor+average_percent_over/100.0)) {
-					sprintf(log_buffer,
-						"ncpus %.2f exceeded limit %lu (sum)",
-						(double)cput_sum/(double)walltime_sum,
-						value);
-					if (cpuaverage) { /* abort job */
-						return (TRUE);
-					} else if ((pjob->ji_qs.ji_svrflags & JOB_SVFLG_cpuperc) == 0) {
-						/* just log it */
-						log_event(PBSEVENT_JOB,
-							PBS_EVENTCLASS_JOB, LOG_INFO,
-							pjob->ji_qs.ji_jobid,
-							log_buffer);
-						pjob->ji_qs.ji_svrflags |= JOB_SVFLG_cpuperc;
+			if (preswalltime != NULL) {
```

Comment:
Also check rs_value.at_flags & ATR_VFLAG_SET and then only assign the value to preswalltime.